demux/utils/indexreport.py

```diff
@@ -0,0 +1,136 @@
+        cluster_counts: int,
+        INDEX_REPORT_HEADER: list,
+        log
+    ):
+        self.INDEX_REPORT_HEADER = INDEX_REPORT_HEADER
+        self.out_dir = out_dir
+        self.index_report_path = index_report_path
+        self.flowcell_id = flowcell_id
+        self.cluster_counts = cluster_counts
+        self.log = log
+
+    def _html_content(self):
+        """Get the content of the report"""
+
+        html_content = bs4.BeautifulSoup(self.index_report_path, "html.parser")
+        self.html_content: bs4.BeautifulSoup = html_content
+
+    def _report_tables(self):
+        """Get the ReportTables inside the html report"""
+
+        report_tables = self.html_content.find_all("table", id="ReportTable")
+        self.report_tables: bs4.ResultSet = report_tables
+
+    def _header_samples_table(self):
+        """Get the header from the large table with all sample clusters"""
+
+        headers = []
+        header_index = {}
+
+        index_counter = 0
+
+        tmp_headers = self.report_tables[1].tr.find_all("th")
+
+        for column in tmp_headers:
+            column = re.sub("<br/>", "", str(column))
+            column = re.sub("<.*?>", "", column)
+            headers.append(column)
+            header_index[column] = index_counter
+            index_counter += 1
+
+        self.raw_headers = self.report_tables[1].tr
+        self.headers_report_tables: list = headers
+        self.header_index: dict = header_index
+
+    def _top_unknown_barcodes_table(self):
+        """Get the table with the top unknown barcodes"""
+
+        self.top_unknown_barcodes = self.report_tables[2]
+
+    def _get_low_cluster_counts(self):
+        """Find samples with low cluster counts"""
+
+        low_cluster_counts = []
+
+        for row in self.report_tables[1].find_all("tr")[1:]:
+            tmp_row = row.find_all("td")
+            tmp_project = re.sub(
+                "<.*?>", "", str(tmp_row[self.header_index["Project"]])
+            )
+            tmp_cluster_count = re.sub(
+                "<.*?>", "", str(tmp_row[self.header_index["PF Clusters"]])
+            )
+            if tmp_project != "indexcheck":
+                if int(tmp_cluster_count.replace(",", "")) < self.cluster_counts:
+                    low_cluster_counts.append(row)
+
+        self.low_cluster_counts: list = low_cluster_counts
+
+    def parse_report(self):
+        """Parses bcl2fastq indexcheck report"""
+
+        self.log.info(
+            f"Parsing file {self.index_report_path.name}, extracting top unkown barcodes and samples with cluster"
+            f"counts lower than {self.cluster_counts}"
+        )
+
+        self._html_content()
+        self._report_tables()
+        self._header_samples_table()
+        self._top_unknown_barcodes_table()
+        self._get_low_cluster_counts()
+
+    def write_summary(self):
+        """Compile a summary report of the bcl2fastq report"""
+
+        out_dir_path = Path(self.out_dir)
```

Comment:
So the sane default would be de demux directory, which I would use when calling the script in our demux script. The reason I cannot set a good default is because I feel uncertain about wether the structure of the demux out directories will remain consistent from bcl2fastq updates